### PR TITLE
feat(api): run Prettier after templates generation

### PR DESCRIPTION
**Pull request by francoischalifour**
A common issue with dynamic values injected in the templates is that generated files were not always Prettier-compliant. For instance, if an attribute name or an index name are longer than anticipated, they should be on multiple lines.

This PRs aims a solving this issue by running Prettier at the `teardown` hook. Prettier is run from the Create InstantSearch App dependency, but run with the templates local `.prettierrc` configuration. This therefore respects the configuration specified in the template.

You can see the results in the snapshots generated.

### CODE REVIEW

File: src/tasks/node/teardown.js [hunk 1]
```diff
@@ -1,18 +1,34 @@
...
-      const currentDirectory = process.cwd();
-      const cdPath =
-        path.join(currentDirectory, config.name) === config.path
-          ? config.name
-          : config.path;
+      const command = hasYarn ? 'yarn' : 'npm';
+      const installCommand = `${command} install`;
+      const startCommand = `${command} start`;
 
       console.log();
```
Comment (tkrugg) [inline src/tasks/node/teardown.js]: unrelated but is this empty console.log really useful?

Comment (francoischalifour) [reply src/tasks/node/teardown.js]: Yes – it adds a new line in the UI.

**Comment by francoischalifour** [comment]
So that it works for users using CISA with `yarn create` or `npx`, should we consider `prettier` as part of `dependencies` instead of `devDependencies` now?

File: src/tasks/node/teardown.js [hunk 2]
```diff
@@ -1,18 +1,34 @@
...
+  const cdPath =
+    path.join(currentDirectory, config.name) === config.path
+      ? config.name
+      : config.path;
+
+  try {
+    const command = hasYarn ? 'yarn' : 'npx';
+
+    execSync(
+      `${command} prettier "${cdPath}/src/**/*.{json,html,css,js,vue,ts,tsx}" --write --config "${cdPath}/.prettierrc"`,
```
Comment (Haroenv) [inline src/tasks/node/teardown.js]: yes, this makes prettier a dependency.

Comment (Haroenv) [inline src/tasks/node/teardown.js]: is the config always available, and does prettier not take that one by default?

Comment (francoischalifour) [reply src/tasks/node/teardown.js]: Yes, the config is available for all our JavaScript templates. If this turns out not to be true, we still swallow the error.

Nope, it takes the "global" one from CISA by default.

File: src/templates/InstantSearch.js/src/app.js.hbs [hunk 3]
```diff
@@ -25,15 +25,15 @@ search.addWidget(
...
-        <article>
-          <h1>\{{#helpers.highlight}}{ "attribute": "{{attributesToDisplay.[0]}}" }\{{/helpers.highlight}}</h1>
-          {{#each attributesToDisplay}}
-          {{#unless @first}}
-          <p>\{{#helpers.highlight}}{ "attribute": "{{this}}" }\{{/helpers.highlight}}</p>
-          {{/unless}}
-          {{/each}}
-        </article>
-      `,
+<article>
```
Comment (Haroenv) [inline src/templates/InstantSearch.js/src/app.js.hbs]: Why did you choose to put this without indentation?

Comment (francoischalifour) [reply src/templates/InstantSearch.js/src/app.js.hbs]: That's how Prettier formats it and it's easier to edit it so I figured why not. No strong opinion though.

**Pull request merged**